.github/workflows/gh-pages.yml

```diff
@@ -6,26 +6,40 @@ on:
 
 jobs:
 
-  documentation-job:
+  build-documentation:
     runs-on: ubuntu-24.04
-
+    permissions:
+      contents: read
     steps:
       - name: SCM Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Setup Python & Poetry Environment
-        uses: exasol/python-toolbox/.github/actions/python-environment@1.1.0
+        uses: exasol/python-toolbox/.github/actions/python-environment@1.4.0
 
       - name: Build Documentation
         run: |
           poetry run -- nox -s docs:multiversion
+          rm -r .html-documentation/*/.doctrees
 
-      - name: Deploy
-        uses: JamesIves/github-pages-deploy-action@v4.7.3
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
         with:
-          branch: gh-pages
-          folder: .html-documentation
-          git-config-name: Github Action
-          git-config-email: opensource@exasol.com
+          path: .html-documentation
+
+  deploy-documentation:
```

Comment:
I updated the gh-pages settings to use the action (not the branch) & allow from tag.